src/test/java/io/swagger/oas/test/examples/ResponseExamplesTest.java

```diff
@@ -117,4 +132,193 @@ public void testResponseYamlExample(@Injectable final List<io.swagger.v3.parser.
     private void assertEqualsIgnoreLineEnding(String actual, String expected) {
         assertEquals(actual.replace("\n", ""), expected);
     }
+
+    public void testRandomJsonExample(@Injectable final List<io.swagger.v3.parser.core.models.AuthorizationValue> auths) throws Exception {
+        Configuration config = new Configuration();
+        List<String> exampleProcessor = new ArrayList<>();
+        exampleProcessor.add("random");
+        config.setExampleProcessors(exampleProcessor);
+        ParseOptions options = new ParseOptions();
+        options.setResolveFully(true);
+
+        OpenAPI openAPI = new OpenAPIV3Parser().readLocation( "src/test/swagger/oas3.yaml",auths, options).getOpenAPI();
+        Operation operation = openAPI.getPaths().get( "/mockResponses/objectMultipleExamples").getGet();
+
+        OpenAPIOperationController controller = new OpenAPIOperationController(
+                config, "/mockResponses/objectMultipleExamples", "GET", operation, openAPI.getComponents().getSchemas() );
+
+        ContainerRequestContext requestContext = mock(ContainerRequestContext.class);
+        UriInfo uriInfo = mock( UriInfo.class );
+
+        stub( uriInfo.getPath()).toReturn( "/mockResponses/responseWithExamples");
+        stub( uriInfo.getQueryParameters()).toReturn( new MultivaluedHashMap<String, String>());
+        stub( uriInfo.getPathParameters()).toReturn( new MultivaluedHashMap<String, String>());
+
+        stub( requestContext.getAcceptableMediaTypes()).toReturn(Arrays.asList(MediaType.APPLICATION_JSON_TYPE));
```

Comment:
i'm wondering what happen if the content type is not any of the expected. if an error must be thrown or a response with a specific status code, can you add a test to verify the expected behave for that scenario?

Reply:
oki

Reply:
We have two test here. We changed the logic, now we won`t throw an exception, but send the response with the first mediaType of the response, (it works like that already)
here are the tests for that

https://github.com/swagger-api/swagger-inflector/blob/2.0/src/test/java/io/swagger/oas/test/integration/responses/SchemaResponseIT.java#L98-L123